### PR TITLE
MSDKUI-1735: Add possibility to insert fixed coordinates

### PR DESCRIPTION
**Pull request by ants86**
- Added functionality to use specific coordinates as destination.
- Adapted the tests.
- Minor improvements.

Signed-off-by: Ants Peetsalu <43877908+ants86@users.noreply.github.com>

### CODE REVIEW

File: MSDKUI_Demo_UI_Tests/Impl/Views/Waypoints/WaypointActions.swift [hunk 1]
```diff
@@ -92,4 +93,22 @@ enum WaypointActions {
             }
         )
     }
+
+    /// Sets current map view to new coordinates.
+    ///
+    /// - Parameter mapData: set of latitude and longitude coordinates.
```
Comment (OguzHere) [inline MSDKUI_Demo_UI_Tests/Impl/Views/Waypoints/WaypointActions.swift]: "set" should be "Set".

File: MSDKUI_Demo_UI_Tests/Tests/DriveNavigation/Functional/GuidanceAndManeuversTests.swift [hunk 2]
```diff
@@ -204,7 +203,7 @@ final class GuidanceAndManeuversTests: XCTestCase {
             DriveNavigationActions.waitForArrival()
 
             // Verify that ETA checks were run during simulation
-            GREYAssertTrue(DriveNavigationActions.etaCheckCounter >= 4,
+            GREYAssertTrue(DriveNavigationActions.etaCheckCounter >= 1,
                            reason: "ETA data checks must be run at least 4 times")
```
Comment (OguzHere) [inline MSDKUI_Demo_UI_Tests/Tests/DriveNavigation/Functional/GuidanceAndManeuversTests.swift]: Is "at least 4 times" still valid?

File: MSDKUI_Demo_UI_Tests/Impl/Core/CoreActions.swift [hunk 3]
```diff
@@ -93,39 +92,61 @@ enum CoreActions {
...
+    static func centerOfElement(element: GREYMatcher) -> CGPoint {
+        var point = CGPoint(x: 0.0, y: 0.0)
+        EarlGrey.selectElement(with: element).perform(
+            GREYActionBlock.action(withName: "Get center point of an UI element") { element, errorOrNil in
+                guard
+                    errorOrNil != nil,
+                    let element = element as? UIView else {
+                        return false
+                }
+                let heightMiddle = element.bounds.height / 2
```
Comment (OguzHere) [inline MSDKUI_Demo_UI_Tests/Impl/Core/CoreActions.swift]: element.frame vs element.bounds? If bounds, then centerOfElementBounds()?

File: MSDKUI_Demo_UI_Tests/Impl/Views/Waypoints/WaypointActions.swift (hunk 1 above)
Comment (ants86) [reply MSDKUI_Demo_UI_Tests/Impl/Views/Waypoints/WaypointActions.swift]: Correct, will change

File: MSDKUI_Demo_UI_Tests/Tests/DriveNavigation/Functional/GuidanceAndManeuversTests.swift (hunk 2 above)
Comment (ants86) [reply MSDKUI_Demo_UI_Tests/Tests/DriveNavigation/Functional/GuidanceAndManeuversTests.swift]: No, routes are now shorter hence only 1 check, will update doc.

File: MSDKUI_Demo_UI_Tests/Impl/Core/CoreActions.swift (hunk 3 above)
Comment (ants86) [reply MSDKUI_Demo_UI_Tests/Impl/Core/CoreActions.swift]: Yours is better, done.

Comment (otaviohere) [reply MSDKUI_Demo_UI_Tests/Impl/Core/CoreActions.swift]: **Bounds** is how a view sees itself. **Frame** is how others see the view. In other words, the main different between bound and frame is the `.origin`, not the `.size`.

So if all you want is the scalar (magnitude, not direction) for 1/2 the height and 1/2 the width, then it doesn't matter if bounds or frame is used.

If you want the center point relative the the view's origin, then you can create the point using `x = 1/2 * view.bounds.width` and `y = 1/2 * view.bounds.height`.

But if you want the center point to relative to the superview's origin, then `x = view.frame.x + 1/2 * view.bounds.width` and `y = view.frame.y + 1/2 * view.bounds.height`.

**Pull request merged**